src/base/materialized.ts

```diff
@@ -230,60 +236,203 @@ export class MaterializedChatMessage implements IMaterializedNode {
 		return tokenizer.countMessageTokens({ ...this.toChatMessage(), content: '' });
 	});
 
-	private readonly _text = once(() => {
-		let result = '';
+	private readonly _text = once((): (string | MaterializedChatMesageImage)[] => {
+		let result: (string | MaterializedChatMesageImage)[] = [''];
 		for (const { text, isTextSibling } of textChunks(this)) {
+			if (text instanceof MaterializedChatMesageImage) {
+				result.push(text);
+				for (const child of text.children) {
+					if (child instanceof MaterializedChatMessageTextChunk) {
+						result.push(child.text);
+					}
+				}
+				break;
+			}
 			if (
 				text.lineBreakBefore === LineBreakBefore.Always ||
 				(text.lineBreakBefore === LineBreakBefore.IfNotTextSibling && !isTextSibling)
 			) {
-				if (result.length && !result.endsWith('\n')) {
-					result += '\n';
+				let res = result[result.length - 1]
+				if (!(res instanceof MaterializedChatMesageImage) && result.length && !res.endsWith('\n')) {
```

Comment:
nit: this reads a little better imo, because we want to see if there's string content before this message, rather than a 'not-image' before it

```suggestion
				let prev = result[result.length - 1]
				if (typeof prev === 'string' && result.length && !prev.endsWith('\n')) {
```